posts/2020/csv-downloader-in-react.md

```diff
@@ -0,0 +1,41 @@
+.. title: CSV Downloader in React
+.. slug: csv-downloader-in-react
+.. date: 2020-07-17 13:17:20 UTC+05:30
+.. tags: ReactJS
+.. link:
+.. description:
+.. type: text
+.. summary: In this post, we will check how the UI data can be downloaded in CSV format. This functionality can be achieved on the back-end (Django REST) itself using 'Content-Disposition' header but we will prefer front-end approach for implementation and also explain why so.
```

Comment:
`Content-Disposition` alone does not work. If that worked, there is no reason to implement this in front-end. So what is the problem with `Content-Disposition` needs to be explained. 

Often in any application, it's a requirement in front-end to download certain dynamic data from the server as a file. Typically when such requests are done using Ajax, setting simple `Content-Disposition` alone does not work. In this post we'll look at some details of things that the frontend should take care of in ReactJS setup.